Source: upstream, SVN rev. 18090

Fix the test suite - /var/run/net-snmp does not exist at the time
'make test' is running. So we need another place to put temporary
files - let it be /tmp. SELinux allows it, snmpd during tests is
unconfined.

diff -up net-snmp-5.5/testing/tests/T059trapdtraphandle.orig net-snmp-5.5/testing/tests/T059trapdtraphandle
--- net-snmp-5.5/testing/tests/T059trapdtraphandle.orig	2007-12-19 23:31:28.000000000 +0100
+++ net-snmp-5.5/testing/tests/T059trapdtraphandle	2010-01-29 19:57:30.000000000 +0100
@@ -22,6 +22,7 @@ snmp_version=v2c
 TESTCOMMUNITY=testcommunity
 
 CONFIGTRAPD [snmp] persistentDir $SNMP_TMP_PERSISTENTDIR
+CONFIGTRAPD [snmp] tempFilePattern /tmp/snmpd-tmp-XXXXXX
 CONFIGTRAPD authcommunity execute $TESTCOMMUNITY
 CONFIGTRAPD doNotLogTraps true
 CONFIGTRAPD traphandle default $0 $1 traphandle
